packages/@react-spectrum/tabs/src/Tabs.tsx

```diff
@@ -226,8 +231,13 @@ function TabLine(props: TabLineProps) {
       }
       setStyle(styleObj);
     }
+  }, [direction, setStyle, selectedTab, orientation]);
+
+  useLayoutEffect(() => {
+    onResize();
+  }, [onResize, direction, scale, selectedKey, orientation, selectedTab?.offsetLeft]);
```

Comment:
Think we can get rid of some of these extra deps since they are already in the callback's dep array
```suggestion
  }, [onResize, scale, selectedKey, selectedTab?.offsetLeft]);
```

Reply:
I don't think so—from my understanding, the `useCallback` deps gives us a memoized callback function with non-stale values. And the `useLayoutEffect` deps ensure we *call* that resize function any time its deps change.

Reply:
@LFDanLu is correct, the onResize function will change anytime its dep array changes (as you said, it's memoized based on that dep array), which will mean that it changes in this dependency array and cause the useLayoutEffect to run

however, I have a different concern. we still have offsetLeft in this dependency array, which means we're getting lucky and render happens to be running after some change in the browser rendered dom. we should never have a dom rendered measurement as a direct reference in our dependency arrays, it's only safe to read dom values during effects. Here's an overly simplified version of why, https://codesandbox.io/s/rough-butterfly-98ny0r?file=/src/App.js

Instead, we need to be figuring out what actually changed (a resize event?) and hooking up to that to get offsetLeft into state. This will ensure a re-render and therefore the line will be placed correctly. We may need to move the state up a component in order to accomplish this. I haven't looked into it, just noticed it when I scanned the review.

Reply:
Ah, yep! Not sure how I looked right over onResize being in the second dep array, haha.

I think I'm following your other concern. Can take another look sometime soon, though likely not until after holiday.

Reply:
That's fine, no rush. I've been trying to find time to look into it as well.